int64 key
float32[3] vector
float32[3] sigmas